community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/CypherCompiler.scala

```diff
@@ -88,9 +83,11 @@ class CypherCompiler(graph: GraphDatabaseService,
 
   @throws(classOf[SyntaxException])
   def preParseQuery(queryText: String): PreParsedQuery = {
+    val logger = new RecordingNotificationLogger
```

Comment:
The notifications are always recorded but only ExplainExecutionResult returns the actual notifications, the other implementations returns empty iterators.